### DISCUSSION TITLE
mixing theme variants

### DISCUSSION THREAD

**Discussion by chiptus** [tailwindlabs/tailwindcss#9013] [Help]
in our codebase, we have two editions (community and business), and we have different themes (light/dark/high-contrast), so I have the following variants:
```js
addVariant('be', '[data-edition="BE"] &');
addVariant('th-highcontrast', '[theme="highcontrast"] &');
addVariant('th-dark', '[theme="dark"] &');
```

my problem is doing something like `be:th-dark:bg-black`. I want it to be `[data-edition="BE"][theme="dark"]`, while it actually adds a space between them. this is reasonable as that's how I defined the variant. removing the space before `&` will cause bugs in other places where I use only one variant.

do you have a suggestion for mixing variants?

my solutions would be either to introduce a variant like `be-th-dark`, or to move the theme attribute to the body element, and then need to make sure it's always after the be variant

**Comment by wongjn** [comment]
You could use [`:is()`](https://developer.mozilla.org/ja/docs/Web/CSS/:is) so that the variants combine without spaces and thus can be chained whichever order:

```js
addVariant('be', '&:is([data-edition="BE"] *)');
addVariant('th-highcontrast', '&:is([theme="highcontrast"] *)');
addVariant('th-dark', '&:is([theme="dark"] *)');
```

[Here's a playground of it in action](https://play.tailwindcss.com/g4h699R0eD).

**Comment by chiptus** [comment]
Thanks!
